### PR TITLE
Prevent purchase of cancelled tix

### PR DESCRIPTION
**Pull request by mikethetike**
### References Issues:
Related to big-neon/bn-web#1620

### Description:
Prevent purchase of tickets for cancelled events and ticket types

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

File: db/src/models/ticket_types.rs [hunk 1]
```diff
@@ -584,14 +584,16 @@ impl TicketType {
...
-    pub fn is_event_not_draft(
+    pub fn is_event_available_for_sale(
         ticket_type_id: &Uuid,
         conn: &PgConnection,
     ) -> Result<bool, DatabaseError> {
         let valid_ticket_count: i64 = ticket_types::table
             .inner_join(events::table)
             .filter(ticket_types::id.eq(ticket_type_id))
             .filter(events::status.ne(EventStatus::Draft.to_string()))
+            .filter(events::status.ne(EventStatus::Draft.to_string()))
```
Comment (Krakaw) [inline db/src/models/ticket_types.rs]: Is this a duplicate?
```suggestion
            .filter(events::cancelled_at.is_null())
```

Comment (mikethetike) [reply db/src/models/ticket_types.rs]: Good catch. Must have copied/pasted and then not checked

**Comment by Krakaw** [comment]
This is also Connected to big-neon/bigneon#192

**Comment by Anziskavds** [comment]
Related to #192

Tested on Beta and I was unable to buy tickets and received an error message saying Event not published but this needs to be fixed as per the stated requirements. Issue can not be closed until it is fixed as per the requirements.

**Pull request merged**